ddl/db_change_test.go

```diff
@@ -223,8 +281,7 @@ func (s *testStateChangeSuite) test(c *C, tableName, alterTableSQL string, testI
 	err = testInfo.execSQL(3)
 	c.Assert(err, IsNil)
 	c.Assert(errors.ErrorStack(checkErr), Equals, "")
-	callback = &ddl.TestDDLCallback{}
-	d.(ddl.DDLForTest).SetHook(callback)
+	d.(ddl.DDLForTest).SetHook(originalCallback)
```

Comment:
use defer maybe better. If the test failed and return without reset the hook, May cause other test fail.